tests/unit/fem/test_sparse_matrix.cpp

```diff
@@ -0,0 +1,202 @@
+// Copyright (c) 2010-2020, Lawrence Livermore National Security, LLC. Produced
+// at the Lawrence Livermore National Laboratory. All Rights reserved. See files
+// LICENSE and NOTICE for details. LLNL-CODE-806117.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability visit https://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the BSD-3 license. We welcome feedback and contributions, see file
+// CONTRIBUTING.md for details.
+
+#include "catch.hpp"
+#include "mfem.hpp"
+#include <fstream>
+#include <iostream>
+
+using namespace mfem;
+
+#ifdef MFEM_USE_MPI
+
+namespace sparse_matrix_test
+{
+
+double coeff_function(const Vector &x)
+{
+   return 1.0 + x[0]*x[0];
+}
+
+void velocity_function(const Vector &x, Vector &v)
+{
+   int dim = x.Size();
+   switch (dim)
+   {
+      case 1: v(0) = 1.0; break;
+      case 2: v(0) = x(1); v(1) = -x(0); break;
+      case 3: v(0) = x(1); v(1) = -x(0); v(2) = x(0); break;
+   }
+}
+
+enum class Coeff {Const, Grid, Quad};
+
+static std::string getString(Coeff coeff_type)
+{
+   switch (coeff_type)
+   {
+      case Coeff::Const:
+         return "Const";
+         break;
+      case Coeff::Grid:
+         return "Grid";
+         break;
+      case Coeff::Quad:
+         return "Quad";
+         break;
+   }
+   mfem_error("Unknown CeedCoeff.");
+   return "";
+}
+
+enum class Problem {Mass, Convection, Diffusion};
+
+static std::string getString(Problem pb)
+{
+   switch (pb)
+   {
+      case Problem::Mass:
+         return "Mass";
+         break;
+      case Problem::Convection:
+         return "Convection";
+         break;
+      case Problem::Diffusion:
+         return "Diffusion";
+         break;
+   }
+   mfem_error("Unknown Problem.");
+   return "";
+}
+
+void test_sparse_matrix(const char* input, int order, const Coeff coeff_type,
+                        const Problem pb, const bool keep_nbr_block)
+{
+   std::string knb = keep_nbr_block ? "ON" : "OFF";
+   std::string section = "keep_nbr_block: " + knb + "\n" +
+                         "coeff_type: " + getString(coeff_type) + "\n" +
+                         "pb: " + getString(pb) + "\n" +
+                         "order: " + std::to_string(order) + "\n" +
+                         "mesh: " + input;
+   INFO(section);
+   Mesh mesh(input, 1, 1);
+   mesh.EnsureNodes();
+   mesh.UniformRefinement();
+   ParMesh pmesh(MPI_COMM_WORLD, mesh);
+   int dim = mesh.Dimension();
+
+   FiniteElementCollection *fec;
+   if (pb == Problem::Convection)
+   {
+      fec = new L2_FECollection(order, dim, BasisType::GaussLobatto);
```

Comment:
Could we also add a test for ``BasisType::Positive`` ?

Reply:
done.